codepropertygraph/src/main/scala/io/shiftleft/utils/StringInterner.scala

```diff
@@ -0,0 +1,34 @@
+package io.shiftleft.utils
+
+import gnu.trove.map.hash.THashMap
+
+trait StringInterner {
```

Comment:
Please add some documentation about what this does and why we have it. 
This is especially important because there are no tests - I wouldn't know how to test this either, so I guess just add docs so that the next person is less likely to throw this out.

Reply:
@mpollmeier I've added some docs that should hint why it is "Strong" instead of "String" :)
My plan was to have "WeakInterner" in addition to the strong one, that will hold weak refs for use cases where it is not good to keep refs to all the values inside a hashmap.

Reply:
ah, hehe ok. since we only have one version, why not make it a class with a normal constructor, rather than a `make` function? anyway, no strong feelings, you can leave it as is if you prefer.

Reply:
@mpollmeier I have no strong feelings about this but I prefer to have factory method, so when you provide new implementation with different type of hashmap or some other optimization, you can easily change it in one place and try it out.

Reply:
ok